tests/main/snapctl/task.yaml

```diff
@@ -55,12 +55,20 @@ restore: |
 
 execute: |
     echo "Run the hook that calls snapctl"
-    export SNAP_CONTEXT="foo" # Just enough to hit the API
-    if ! snap run --hook=apply-config snapctl-hooks; then
-        echo "Failed to run hook"
+
+    # This context is enough to hit the API, but the hook will fail since this
+    # context is obviously invalid. That failure still means we're correctly
+    # hitting the API, though, which is all we care about here.
+    export SNAP_CONTEXT="foo"
+    if output="$(snap run --hook=apply-config snapctl-hooks)"; then
+        echo "Expected the hook to fail"
         exit 1
     fi
 
+    if [[ ! "$output" =~ .*"no context for ID: \"foo\"".* ]]; then
```

Comment:
Ah, and the problem is actually on the former line:
```
error: cannot run snapctl: cannot run snapctl: no context for ID: "foo"
+ output='Running snapctl. It should be in my PATH and not cause denials.'
+ [[ ! Running snapctl. It should be in my PATH and not cause denials. =~ .*no context for ID: "foo".* ]]
```

Need something like `output="$(snap ... 2>&1 >/dev/null)"` to capture only stderr.